src/migration/mod.rs

```diff
@@ -0,0 +1,341 @@
+
+        for item in items {
+            del_ids.push(item.id);
+            let ins = sqlx::query(
+                "INSERT INTO items (profile_id, kind, category, name, value)
+            VALUES (1, 2, ?1, ?2, ?3)",
+            )
+            .bind(item.category)
+            .bind(item.name)
+            .bind(item.value)
+            .execute(&mut self.conn)
+            .await?;
+            let item_id = ins.last_insert_rowid();
+            for (plain, name, value) in item.tags {
+                sqlx::query("INSERT INTO items_tags (item_id, plaintext, name, value) VALUES (?1, ?2, ?3, ?4)")
+                .bind(item_id)
+                .bind(plain)
+                .bind(name)
+                .bind(value)
+                .execute(&mut self.conn)
+                .await?;
+            }
+        }
+        sqlx::query("DELETE FROM items_old WHERE id IN (?1)")
+            .bind(
+                del_ids
+                    .iter()
+                    .map(u32::to_string)
+                    .collect::<Vec<String>>()
+                    .join(","),
+            )
+            .execute(&mut self.conn)
+            .await?;
+        Ok(())
+    }
+
+    async fn create_config(&mut self, indy_key: &IndyKeyWithMasterAndSalt) -> Result<(), Error> {
+        let salt_hex = hex::encode(&indy_key.salt);
+        let pass_key = format!("kdf:argon2i:13:mod?salt={salt_hex}");
+
+        sqlx::query("INSERT INTO config (name, value) VALUES (?1, ?2)")
+            .bind("default_profile")
+            .bind(&self.wallet_name)
+            .execute(&mut self.conn)
+            .await?;
+
+        sqlx::query("INSERT INTO config (name, value) VALUES (?1, ?2)")
+            .bind("key")
+            .bind(pass_key)
+            .execute(&mut self.conn)
+            .await?;
+
+        Ok(())
+    }
+
+    async fn insert_profile(&mut self, key: &[u8]) -> Result<(), Error> {
+        sqlx::query("INSERT INTO profiles (name, profile_key) VALUES (?1, ?2)")
+            .bind(&self.wallet_name)
+            .bind(key.to_vec())
+            .execute(&mut self.conn)
+            .await?;
+
+        Ok(())
+    }
+
+    async fn fetch_pending_items<
+        T: Send + Unpin + for<'r> sqlx::FromRow<'r, sqlx::sqlite::SqliteRow>,
+    >(
+        &mut self,
+        limit: u8,
+    ) -> Result<Option<Vec<T>>, Error> {
+        let res = sqlx::query_as(
+            "SELECT i.id, i.type, i.name, i.value, i.key,
+        (SELECT GROUP_CONCAT(HEX(te.name) || ':' || HEX(te.value))
+            FROM tags_encrypted te WHERE te.item_id = i.id) AS tags_enc,
+        (SELECT GROUP_CONCAT(HEX(tp.name) || ':' || HEX(tp.value))
+            FROM tags_plaintext tp WHERE tp.item_id = i.id) AS tags_plain
+        FROM items_old i LIMIT ?1",
+        )
+        .bind(limit)
+        .fetch_all(&mut self.conn)
+        .await?;
+
+        match res.len() {
+            0 => Ok(None),
+            _ => Ok(Some(res)),
+        }
+    }
+}
+
+#[cfg(test)]
```

Comment:
This test only works locally, so I will remove it when we merge it or we need to find a way to create a simple indy-sdk wallet.

Reply:
Maybe you can push your local indy-wallet and use that to run the tests? 

Then we ofc. need to have a way regenerate it in some way, but I don't suspect we'd need to change the input very often

Reply:
Yeah I think that would be fine.